include/swift/SIL/SILFunction.h

```diff
@@ -350,6 +362,9 @@ class SILFunction
   /// The function is in a statically linked module.
   unsigned IsStaticallyLinked : 1;
 
+  /// The function is a witness to an ad-hoc requirement.
+  unsigned IsAdHocRequirementWitness : 1;
```

Comment:
Do you even need this flag here, the check could be against pointer `AdHocRequirementFunction` which is _current_ not even set when `setDistributedAdHocRequirementFunction` is called...

Reply:
I think you need to do the same thing as `ReplacedFunction` where it's mangling is serialized into the attribute - https://github.com/apple/swift/blob/main/lib/SIL/Parser/ParseSIL.cpp#L1072-L1092

Reply:
Hmm yeah very possible that I don't need it -- might have overdone the "same as `dynamic`", thanks I didn't spot that piece in ParseSIL for the replaced function!

Reply:
No worries!